docs: use DocumenterCitations

rfourquet opened the pull request:
> [DocumenterCitations.jl](https://github.com/ali-ramadhan/DocumenterCitations.jl) uses Bibliography.jl to add support for BibTeX citations and references in documentation pages generated by Documenter.jl.

Cf. #311.
This PR adds a dummy reference in a bib file, which is referenced from a docstring or from the documentation in an .md file. 
If someone wants to provide a real citation (maybe @wdecker or @GDeFranceschi have one?), I will replace the dummy one.

rfourquet committed:
docs: use DocumenterCitations

GDeFranceschi commented:
> > [DocumenterCitations.jl](https://github.com/ali-ramadhan/DocumenterCitations.jl) uses Bibliography.jl to add support for BibTeX citations and references in documentation pages generated by Documenter.jl.
> 
> Cf. #311.
> This PR adds a dummy reference in a bib file, which is referenced from a docstring or from the documentation in an .md file.
> If someone wants to provide a real citation (maybe @wdecker or @GDeFranceschi have one?), I will replace the dummy one.

A reference I will use for the documentation will surely be:
Taylor, D. E., Pairs of Generators for Matrix Groups. I, The Cayley Bulletin, 3 (1987). 
Anyway, I will add most of my references once the branches involving weird algorithms are merged to the `master` branch.

fingolfin commented:
@GDeFranceschi can you provide a bibtex version of that ref? Alternatively I guess mathscinet can produce bibtex

GDeFranceschi commented:
I did not find the article on MathSciNet. I produced something like this:

> @article
>{Tay87,
>    author={D. E. Taylor},
>    title={Pairs of Generators for Matrix Groups. I},
>    journal={The Cayley Bulletin},
>    volume={3},
>    year=1987,
>    pages={76--85}
>}

rfourquet committed:
replace dummy citation by a real one

rfourquet committed:
use absolute path for CitationBibliography

rfourquet commented:
I updated by removing the dummy citations, and added the bib entry provided by @GDeFranceschi . Citations can be referenced via something like `[Tay87](@cite)`.

fingolfin commented:
Thanks, useful! @wdecker this allows you to do bibtex references as you asked for

fingolfin commented:
@GDeFranceschi you can add any other refs now and cite them (e.g. add your thesis!)

GDeFranceschi commented:
Sure, I'll add them in the branch `ManualGroups`.
